docs/concepts.rst

```diff
@@ -116,6 +116,47 @@ DAGs can be used as context managers to automatically assign new operators to th
 
     op.dag is dag # True
 
+.. _concepts:functional_dags:
+
+Functional DAGs
+---------------
+*Added in Airflow 1.10.11*
+
+DAGs can be defined using functional abstractions. Outputs and inputs are sent between tasks using
+:ref:`XCom values <concepts:xcom>`. In addition, you can wrap functions as tasks using the
+:ref:`task decorator <concepts:task_decorator>`. Airflow will also automatically add dependencies between
+tasks to ensure that XCom messages are available when operators are executed.
+
+Example DAG with functional abstraction
+
+.. code:: python
+
+  with DAG(
+      'send_server_ip', default_args=default_args, schedule_interval=None
+  ) as dag:
+
+    # Using default connection as it's set to httpbin.org by default
+    get_ip = SimpleHttpOperator(
+        task_id='get_ip', endpoint='get', method='GET', xcom_push=True
+    )
+
+    @dag.task(multiple_outputs=True)
+    def prepare_email(raw_json: str) -> str:
```

Comment:
This type hint is wrong.

```suggestion
    def prepare_email(raw_json: str) -> dict:
```

Or `Dict[str,str]` if you want to be more precise.